thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
@shinsuke-nara after investigation, I found `getEnclosingClass()` is better. Because both of local class and member class invoke this method return non-null value. 

And I found if the inner class is static class, it doesn't have `this$0` additional field. So if the actionClass is static inner class, it is same as normal class. 

I fixed the logic in the commit fda5e78

References: 
http://docs.oracle.com/javase/tutorial/java/javaOO/nested.html
https://coderanch.com/t/545389/java/Check-class-class